src/FasterDBStore/Store.cs

```diff
@@ -142,22 +143,33 @@ public IEnumerable<(byte[] Key, byte[] Value)> Find(byte table, byte[] prefix)
 
         public void Delete(byte table, byte[] key)
         {
+            var input = default(Input);
+            var output = new Output();
+
             var k = new BufferKey(table, key);
+            session.Delete(ref k, Empty.Default, 0);
+            var status = session.Read(ref k, ref input, ref output, Empty.Default, 0);
 
-            if (db.Delete(ref k, Empty.Default, 0) == Status.PENDING)
+            if (status == Status.PENDING)
             {
-                db.CompletePending(true);
+                session.CompletePending(true);
             }
         }
 
         public void Put(byte table, byte[] key, byte[] value)
         {
+            var input = default(Input);
+            var output = new Output();
+
             var k = new BufferKey(table, key);
             var v = new BufferValue(value);
 
-            if (db.Upsert(ref k, ref v, Empty.Default, 0) == Status.PENDING)
+            session.Upsert(ref k, ref v, Empty.Default, 0);
+            var status = session.Read(ref k, ref input, ref output, Empty.Default, 0);
+
+            if (status == Status.PENDING)
             {
```

Comment:
We don't need to check the Upsert status?

Reply:
we should check the status of upsert and delete, since they are the operation type that can be pending.

Reply:
Before read it? two checks or with one it's enough?

Reply:
I think checking the read status after the upsert/delete operation should be ok